base/tuple.jl

```diff
@@ -172,10 +172,13 @@ end
 isempty(x::Tuple{}) = true
 isempty(x::Tuple) = false
 
-revargs() = ()
-revargs(x, r...) = (revargs(r...)..., x)
-
-reverse(t::Tuple) = revargs(t...)
+reverse(t::Tuple{}) = t
+reverse(t::Tuple{T}) where {T} = t
```

Comment:
Or more simply `t::Tuple{1}` and so on.

Reply:
If I understand it correctly `Tuple{T}` is equivalent to `NTuple{1}`, but for larger number of arguments it would not be equivalent. I add a specialized function for `NTuple{N}` as it ensures type stability for large `N`.

Reply:
Ah, right, I think it needs to be `NTuple{N, Any}` to allow for heterogeneous tuples.  I think that should be equivalent.